proximal/examples/test_deconv_sv_psf.py

```diff
@@ -0,0 +1,49 @@
+# Spatially Varying PSF Deconvolution Algorithm based on: 
+# Flicker & Rigaut, 2005 https://doi.org/10.1364/JOSAA.22.000504
+
+import proximal as px
+from proximal.lin_ops.lin_op import LinOp
+from proximal.lin_ops.black_box import LinOpFactory
+from scipy.signal import fftconvolve
+import numpy as np
+
+############################################################
+
+def my_factory(G, W, n_psf):
+    
+    def forward(im_src, im_sim):
+        assert np.all(im_src.shape == im_sim.shape)
+        im_sim[:] = 0
+        for i in range(n_psf):
+            weight = W[i,:].reshape(*im.shape)
+            psf_mode = G[:,i].reshape(*im.shape)
+            im_sim[:] += fftconvolve(psf_mode, im_src*weight, mode='same')
+        return im_sim
+
+    def adjoint(im_src, im_sim):
+        assert np.all(im_src.shape == im_sim.shape)
+        im_sim[:] = 0
+        for i in range(n_psf):
+            weight = W[i,:].reshape(*im.shape)
+            psf_mode = G[:,i].reshape(*im.shape)
+            im_sim += fftconvolve(im_src, np.flipud(np.fliplr(psf_mode)), mode='same')* weight 
+        return im_sim
+    
+    return LinOpFactory(im.shape, im.shape, forward, adjoint)
+
+def cost_function(x, im, U, W, mu=0.01): 
+    blur_op = my_factory(U, W, 15)
+    im_blur = blur_op(x)
+    data_term = px.sum_squares(im_blur - im) 
+    grad_term = mu*px.norm1(px.grad(im_blur)) + (1-mu)*px.sum_squares(px.grad(im_blur))
```

Comment:
Do you mean `px.grad(x)` instead?

Also, ProxImaL doesn't yet have the `lin_op` duplication feature in order to merge `norm1` and `sum_squares` together. But you can help it out by writing:

```python
im_grad = px.grad(x)
grad_term = mu*px.norm1(im_grad) + (1-mu)*px.sum_squares(im_grad)
```

Reply:
Actually no, I am getting the output from the blur forward model with this command and I want to define the prior on this output. Am I understanding it right? Wouldn't `blur_op(x)` return the processed image?
```Python
im_blur = blur_op(x)
```

Reply:
Thank you for enlightening me on how to best merge the operators. I was actually following the example in `ReadMe` where the following code nests `px.grad(x)` in `px.norm1()`. Isn't our case the same? (copying here from ReadMe)
```Python
# Construct and solve problem.
x = Variable(I.shape)
prob = Problem(sum_squares(x - b/255) + .1 * norm1(grad(x)) + nonneg(x))
prob.solve()
```